cmd/subnetcmd/import.go

```diff
@@ -68,3 +107,101 @@ func importSubnet(cmd *cobra.Command, args []string) error {
 
 	return nil
 }
+
+func importFromAPM() error {
+	installedRepos, err := apmintegration.GetRepos(app)
+	if err != nil {
+		return err
+	}
+
+	customRepo := "Download new repo"
+	installedRepos = append(installedRepos, customRepo)
+
+	promptStr := "What repo would you like to import from"
+	repo, err := app.Prompt.CaptureList(promptStr, installedRepos)
+	if err != nil {
+		return err
+	}
+
+	if repo == customRepo {
+		promptStr = "Enter your repo URL"
+		repoURL, err := app.Prompt.CaptureString(promptStr)
+		if err != nil {
+			return err
+		}
+
+		mainBranch := "main"
+		masterBranch := "master"
+		customBranch := "custom"
+		branchList := []string{mainBranch, masterBranch, customBranch}
+		promptStr = "What branch would you like to import from"
+		branch, err := app.Prompt.CaptureList(promptStr, branchList)
+		if err != nil {
+			return err
+		}
+
+		repo, err = apmintegration.AddRepo(app, repoURL, branch)
+		if err != nil {
+			return err
+		}
+
+		err = apmintegration.UpdateRepos(app)
+		if err != nil {
+			return err
+		}
+	}
+
+	subnets, err := apmintegration.GetSubnets(app, repo)
+	if err != nil {
+		return err
+	}
+
+	promptStr = "Select a subnet to import"
+	subnet, err := app.Prompt.CaptureList(promptStr, subnets)
+	if err != nil {
+		return err
+	}
+
+	subnetKey := repo + ":" + subnet
+
+	// Populate the sidecar and create a genesis
+	subnetDescr, err := apmintegration.LoadSubnetFile(app, subnetKey)
+	if err != nil {
+		return err
+	}
+
+	var vmType models.VMType = models.CustomVM
+
+	vmDescr, err := apmintegration.LoadVMFile(app, repo, subnetDescr.VMs[0])
+	if err != nil {
+		return err
+	}
+
+	version := fmt.Sprintf("%d.%d.%d", vmDescr.Version.Major, vmDescr.Version.Minor, vmDescr.Version.Patch)
+
+	sidecar := models.Sidecar{
+		Name:            subnetDescr.Alias,
+		VM:              vmType,
+		VMVersion:       version,
+		Subnet:          subnetDescr.Alias,
```

Comment:
I am not sure, asking instead of going on the code: are we having redundant fields sidecar.Name and sidecar.Subnet?

Reply:
This is to support a future feature where we have multiple VMs per subnet.

Reply:
not sure to understand. Subnet field is used a the CLI subnet name, which is pretty 1-to-1 with a vm. in the future are we going to have multiple vms in a CLI subnet definition?